### DISCUSSION TITLE
How to play local files?

### DISCUSSION THREAD

**Discussion by ninguest** [Androz2091/discord-player#1763] [Q&A] [answered]
Is there a way to play local files with discord-player?

**Comment by ninguest** [comment]
![image](https://github.com/Androz2091/discord-player/assets/71936949/40fde9a4-6a13-4b10-be0a-be70a8765cd4)

I'm not sure how to play local files by adding "search engine value"

**Comment by twlite** [comment, marked as answer]
> Is there a way to play local files with discord-player?

yes absolutely, the procedure is same as usual, you just need to explicitly set `searchEngine` value to `QueryType.FILE`. Here is an example:

```js
const { QueryType } = require('discord-player');

// Relative path also works, but absolute path is recommended
// this is a path to your local file which you want to play
const filePath = 'E:/path/to/music.mp3';

await player.play(channel, filePath, {
    // in order to play local files, we need to explicitly tell discord-player to search that path in our file system
    searchEngine: QueryType.FILE // QueryType.FILE tells discord-player to play from our file system,
    // ... (other options if you need)
});
```

Also, if you are not using `player.extractors.loadDefault()`, make sure to load `AttachmentExtractor` else it will not work. Here is how you can load this extractor:

```js
const { AttachmentExtractor } = require('@discord-player/extractor');

player.extractors.register(AttachmentExtractor, {});
```

**Reply by ninguest** [reply]
Thanks a lot.
